### PR TITLE
Add Plugin wrapper for "Open in New Browser Tab" so it can be disabled.

### PR DESCRIPTION
**Pull request by robertpyke**
## References
Related to issue: https://github.com/jupyterlab/jupyterlab/issues/5274
Also related to my previous PR: https://github.com/jupyterlab/jupyterlab/pull/10193

## Code changes

The "Open in New Browser Tab" logic is effectively another hook to allow for simple download, so we'd like to be able to disable that as well. This change builds on the previous PR, to move that logic into an isolated plugin so that it be disabled. This moves the docmanager and filebrowser components into extensions.

While in the code, I also moved selectors to a common namespace, as they were being repeated in several plugins.

## User-facing changes

None by default:

Context Menu:
Screen Shot 2021-05-28 at 3 33 46 PM

View from new tab (Chrome tab)
Screen Shot 2021-05-28 at 3 43 08 PM

With just the docmanager disabled:

```
jupyter labextension disable @jupyterlab/docmanager-extension:open-browser-tab 
```

Screen Shot 2021-05-28 at 3 47 25 PM

The filebrowser plugin will try to invoke the docmanager plugin, but it won't work (as expected).

But with both disabled (docmanager and filebrowser):

```
jupyter labextension disable @jupyterlab/docmanager-extension:open-browser-tab 
jupyter labextension disable @jupyterlab/filebrowser-extension:open-browser-tab
```

Screen Shot 2021-05-28 at 3 49 13 PM

(now the option isn't shown to Open in Browser Tab).

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @robertpyke :+1:

File: packages/docmanager-extension/src/index.ts [hunk 1]
```diff
@@ -410,14 +410,61 @@ export const downloadPlugin: JupyterFrontEndPlugin<void> = {
...
+ * e.g. jupyter labextension disable @jupyterlab/docmanager-extension:open-browser-tab
+ *
+ * Note: If disabling this, you may also want to disable:
+ * @jupyterlab/filebrowser-extension:open-browser-tab
+ */
+export const openBrowserTabPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/docmanager-extension:open-browser-tab',
+  autoStart: true,
+  requires: [ITranslator, IDocumentManager],
+  optional: [],
```
Comment (jtpio) [inline packages/docmanager-extension/src/index.ts:426]: Looks like this could actually be removed if empty?

```suggestion
```

Comment (robertpyke) [reply packages/docmanager-extension/src/index.ts]: Thanks, I updated the code with the change, then re-ran it, and it appears to be working fine. i.e. Open in new browser tab works as expected with the plugin active.

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**